### DISCUSSION TITLE
Tag filtering

### DISCUSSION THREAD

**Discussion by TaddyJ** [babybuddy/babybuddy#856] [Ideas]
I've seen in some previous posts that the implementation of Tags is going to play a greater role going forward.
I would like to request that the Tags are filtered by the Type of form being completed.
For example I have tags for Bathing/Medicine/feeding/changing/vaccinations etc.
So when I'm in a feeding form I have to pick the tags for feeding out of every tag available. Same goes with changing nappies.
I personally would like it if when tags are added inside a particular form time they only show for that form type.
Id imagine it would be pretty easy to implement.

TLDR: Filter user added tags based on form type.

**Comment by MrApplejuice** [comment]
I think that this is easy to implement, however, it is not necessarily suited for all purposes where you want to use tags: Imagine, for example, a tag for tracking allergic reactions, where you likely would use tag food and diapers to track certain food allergies. All I am saying, is that "global tags" probably need to be present as well, they are the super-set of the feature you are requesting.

So I think this boils down to a UI-centric question of "how to allow creation of topic specific tags in a logical manner". I do not think that this is an unsolvable task. Maybe, dynamic tag creation in the tags-editor can default to global-tags, while the tag-editor page exposes a tag-category option that can be configured, optionally so that tags only appear on certain pages. Something like that?

**Reply by TaddyJ** [reply]
I think a tag management page would be handy. 
I agree some global tags would be good to have. 
The food allergy suggestion tho, my wife and I aren't going to add that tag to the expressing, sleeping or tummy time forms. 
Maybe tags created in notes specifically become global. Tags created in specific forms become form specific?

There is also no easy way to the best of my knowledge to remove tags made in error or redundant tags. So I do think s management page could be useful

**Reply by AlexKalopsia** [reply]
A photo editing tool I use (Capture One) allows to have nested tags defined as "Parent > Child", so you can add the tag "People > Alex" as a tag, and it will parse both tags (”People" and "Alex").

Not sure if it applies to this use case, but maybe a flexible approach
